nexus/src/db/datastore/instance.rs

```diff
@@ -66,22 +69,32 @@ impl DataStore {
 
         let gen = instance.runtime().gen;
         let name = instance.name().clone();
-        let instance: Instance = diesel::insert_into(dsl::instance)
-            .values(instance)
-            .on_conflict(dsl::id)
-            .do_nothing()
-            .returning(Instance::as_returning())
-            .get_result_async(self.pool())
-            .await
-            .map_err(|e| {
+        let project_id = instance.project_id;
+
+        let instance: Instance = Project::insert_resource(
+            project_id,
+            diesel::insert_into(dsl::instance)
+                .values(instance)
+                .on_conflict(dsl::id)
+                .do_nothing(),
+        )
+        .insert_and_get_result_async(self.pool())
+        .await
+        .map_err(|e| match e {
+            AsyncInsertError::CollectionNotFound => Error::ObjectNotFound {
```

Comment:
How is there no authz check here?  If there were, you could use `authz_project.not_found()` instead of making the error by hand here.